libs/remix-lib/src/execution/execution-context.ts

```diff
@@ -23,52 +25,71 @@ if (typeof window !== 'undefined' && typeof window['ethereum'] !== 'undefined')
 */
 
 class StateManagerCommonStorageDump extends StateManager {
-  constructor (arg) {
-    super(arg)
+  keyHashes
+  constructor () {
+    super()
     this.keyHashes = {}
   }
 
-  putContractStorage (address, key, value, cb) {
+  putContractStorage (address, key, value) {
     this.keyHashes[keccak(key).toString('hex')] = bufferToHex(key)
-    super.putContractStorage(address, key, value, cb)
+    return super.putContractStorage(address, key, value)
   }
 
-  dumpStorage (address, cb) {
-    this._getStorageTrie(address, (err, trie) => {
-      if (err) {
-        return cb(err)
-      }
-      const storage = {}
-      const stream = trie.createReadStream()
-      stream.on('data', (val) => {
-        const value = rlp.decode(val.value)
-        storage['0x' + val.key.toString('hex')] = {
-          key: this.keyHashes[val.key.toString('hex')],
-          value: '0x' + value.toString('hex')
-        }
-      })
-      stream.on('end', function () {
-        cb(storage)
+  dumpStorage (address) {
+    return new Promise<StorageDump>((resolve, reject) => {
+      this._getStorageTrie(address).then((trie) => {
+        const storage = {}
+        const stream = trie.createReadStream()
+        stream.on('data', (val) => {
+          const value = rlp.decode(val.value)
+          storage['0x' + val.key.toString('hex')] = {
+            key: this.keyHashes[val.key.toString('hex')],
+            value: '0x' + value.toString('hex')
+          }
+        })
+        stream.on('end', function () {
+          resolve(storage)
+        })
+      }).catch((error) => {
+        reject(error)
       })
     })
   }
 
-  getStateRoot (cb) {
-    const checkpoint = this._checkpointCount
-    this._checkpointCount = 0
-    super.getStateRoot((err, stateRoot) => {
-      this._checkpointCount = checkpoint
-      cb(err, stateRoot)
-    })
+  async getStateRoot (force: boolean = false): Promise<Buffer> {
+    if (!force && this._checkpointCount !== 0) {
+    // throw new Error('Cannot get state root with uncommitted checkpoints')
```

Comment:
do you need to throw here?

Reply:
no, this is why it's commented. 
the original code (this inherits from a ethereumjs-vm class) is throwing but that leads to some inconsistency in our case. and this was not really needed.

Reply:
then why not to remove it? or maybe add a comment why it is there

Reply:
ok it's removed